src/braket/default_simulator/openqasm/_helpers/functions.py

```diff
@@ -98,10 +98,16 @@
             [BooleanLiteral(xv.value ^ yv.value) for xv, yv in zip(x.values, y.values)]
         ),
         getattr(BinaryOperator, "<<"): lambda x, y: ArrayLiteral(
-            x.values[y.value :] + [BooleanLiteral(False) for _ in range(y.value)]
+            x.values[len(y.values) :] + [BooleanLiteral(False) for _ in range(len(y.values))]
+            if isinstance(y, ArrayLiteral)
+            else x.values[y.value :] + [BooleanLiteral(False) for _ in range(y.value)]
         ),
         getattr(BinaryOperator, ">>"): lambda x, y: ArrayLiteral(
-            [BooleanLiteral(False) for _ in range(y.value)] + x.values[: len(x.values) - y.value]
+            [BooleanLiteral(False) for _ in range(len(y.values))]
+            + x.values[: len(x.values) - len(y.values)]
+            if isinstance(y, ArrayLiteral)
+            else [BooleanLiteral(False) for _ in range(y.value)]
+            + x.values[: len(x.values) - y.value]
```

Comment:
[minor] this inline logic is getting hard to read - could it be factored out into an internal helper function? perhaps `casting.py` would be a good place.

Reply:
```
ArrayLiteral: {
        # returns array
        getattr(BinaryOperator, "&"): lambda x, y: ArrayLiteral(
            [BooleanLiteral(xv.value and yv.value) for xv, yv in zip(x.values, y.values)]
        ),
```

Does this mean that these operators can be applied to array literals too now? It might be better to convert `bitstring`s to an `int` type asap where these operations can be handled by python. This would also avoid the possiblity of `{0, 3, 6} & 1` (if indeed this does apply to all array literals and not just bitstring), which to me doesn't make much sense.

Reply:
Also on this theme, it seems as though this is treating `bitstring` literals in the same way as `0b` integer literals ("Integer literals can be written in decimal without a prefix, or as a hex, octal, or binary number, as denoted by a leading 0x/0X, 0o, or 0b/0B prefix. "). I feel as though the 'spirit' of the spec is that the `bit string` is syntactic sugar for an array containing only `0`s and `1`s, rather than a number; but the writing does not specify this.